src/ReadLog.jl

```diff
@@ -0,0 +1,134 @@
+# Read LCM log files directly
+
+# export
+#   LCMlog,
+#   close,
+#   readNextEvent,
+#   handle,
+#   subscribe
+
+# also see JuliaLang's RawFD type, but little information found.
+# www.c4learn/c-programming/c-file-structure-and-file-pointer
+mutable struct FILE
+  level::Cshort
+  token::Cshort
+  bsize::Cshort
+  fd::Cuchar
+  flags::Cuint  # (unsigned flags ;)
+  hold::Cuchar
+  buffer::Ptr{Cuchar}
+  curp::Ptr{Cuchar}
+  istemp::Cuint # (unsigned is temp)
+end
+
+mutable struct lcm_eventlog_t
+  f::Ptr{FILE}
+  eventcount::Clonglong
+end
+const _LCMlog = Ptr{lcm_eventlog_t}
+
+
+struct lcm_eventlog_event_t
+  eventnum::Clonglong
+  timestamp::Clonglong
+  channellen::Cint
+  datalen::Cint
+  channel::Ptr{Cuchar}
+  data::Ptr{Cuchar}
+end
+
+
+function lcm_eventlog_create(file::S where S <: AbstractString)
+  ccall(
+    (:lcm_eventlog_create, LCMCore.liblcm),
+    _LCMlog, #Ptr{lcm_eventlog_t},
+    (Cstring, Cstring),
+    file, "r"
+  )
+end
+
+
+function lcm_eventlog_destroy(_log::Ptr{lcm_eventlog_t})
+  ccall(
+    (:lcm_eventlog_destroy, LCMCore.liblcm),
+    Void,
+    (Ptr{lcm_eventlog_t}, ),
+    _log
+  )
+end
+
+
+function lcm_eventlog_read_next_event(_log::Ptr{lcm_eventlog_t})
+    ccall(
+      (:lcm_eventlog_read_next_event, LCMCore.liblcm),
+      Ptr{lcm_eventlog_event_t},
+      (Ptr{lcm_eventlog_t},),
+      _log
+    )
+end
+
+#
+# Doesn't work yet
+# event = unsafe_wrap(lcm_eventlog_event_t, _event, dim??)
+
+
+
+mutable struct LCMlog
+  _log::_LCMlog
+  subscriptions::Dict{AbstractString, LCMCore.SubscriptionOptions}
+  function LCMlog(filename::S) where {S <: AbstractString}
+    try
+      _log = lcm_eventlog_create(filename)
+      return new(_log, Dict{AbstractString, LCMCore.SubscriptionOptions}())
+    catch e
+      warn(e)
+      backtrace()
+      throw("Cannot open the LCM log file at: $filename")
+    end
+    return new() # should never occur
+  end
+end
+
+
+
+function close(lcmlog::LCMlog)
+  try
```

Comment:
What is the purpose of the `try`-catch here? The `ccall` in `lcm_eventlog_destroy` shouldn't throw a Julia error; rather, I'd expect it to segfault if we try to destroy the same log twice, in which case this `try` won't help.

Reply:
Indeed a segfault. Will use an `if isgood(lcmlog)` instead.

Reply:
Perfect, thank you!